cuda_core/examples/pytorch_example.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+## Usage: pip install "cuda-core[cu12]"
+## python python_example.py
+import sys
+
+import torch
+
+from cuda.core.experimental import Device, LaunchConfig, Program, ProgramOptions, Stream, launch
+
+# SAXPY kernel - passing a as a pointer to avoid any type issues
+code = """
+template<typename T>
+__global__ void saxpy_kernel(const T* a, const T* x, const T* y, T* out, size_t N) {
+ const unsigned int tid = threadIdx.x + blockIdx.x * blockDim.x;
+ if (tid < N) {
+   // Dereference a to get the scalar value
+   out[tid] = (*a) * x[tid] + y[tid];
+ }
+}
+"""
+
+dev = Device()
+dev.set_current()
+
+# Get PyTorch's current stream
+pt_stream = torch.cuda.current_stream()
+print(f"PyTorch stream: {pt_stream}")
+
+
+# Create a wrapper class that implements __cuda_stream__
+class PyTorchStreamWrapper:
+    def __init__(self, pt_stream):
+        self.pt_stream = pt_stream
+
+    def __cuda_stream__(self):
+        stream_id = self.pt_stream.cuda_stream
+        return (0, stream_id)  # Return format required by CUDA Python
+
+
+# Create a wrapper for the PyTorch stream
+pt_stream_wrapper = PyTorchStreamWrapper(pt_stream)
+
+# Initialize a CUDA Python Stream from the PyTorch stream
+s = Stream._init(obj=pt_stream_wrapper)
+print("Successfully created CUDA Python stream from PyTorch stream")
```

Comment:
`Stream._init()` is our internal API. You can just do this with the adaptor:
```suggestion
s = PyTorchStreamWrapper(pt_stream)
```
and `s` will be recognized by cuda-core. In the long term we should have torch's Stream implement `__cuda_stream__()`. I plan to open an issue to track this in a few repos later.